common/wuanalysis/anawu.cpp

```diff
@@ -95,15 +95,89 @@ class WuScope : public CInterface, implements IWuEdge, implements IWuActivity
 };
 
 //-----------------------------------------------------------------------------------------------------------
+typedef enum { WuOptionType_MSec, WuOptionType_Seconds, WuOptionType_Percent } WuOptionType;
+
+struct WuOption
+{
+    const char * name;
+    __int64 value;
+    WuOptionType type;
+};
+
+constexpr struct WuOption wuOptionsDefaults[]
+= { { "minInterestingTime", 10, WuOptionType_MSec},
+    { "minInterestingCost", 30, WuOptionType_Seconds},
+    { "skewThreshold", 20, WuOptionType_Percent} };
+
+class WuAnalyserOptions : public IAnalyserOptions
+{
+public:
+    WuAnalyserOptions()
+    {
+        for (unsigned opt=0; opt<arraysize(wuOptionsDefaults); opt++)
+        {
+            setOptionValue(static_cast<WuOptions>(opt), wuOptionsDefaults[opt].value);
+        }
+    }
+
+    virtual void setOptionValue(WuOptions opt, __int64 val) override
+    {
+        switch(wuOptionsDefaults[opt].type)
+        {
+            case WuOptionType_MSec:
+                wuOptions[opt] = msecs2StatUnits(val);
+                break;
+            case WuOptionType_Seconds:
+                wuOptions[opt] = seconds2StatUnits(val);
+                break;
+            case WuOptionType_Percent:
+                wuOptions[opt] = statSkewPercent((int)val);
+                break;
+        }
+    }
+
+    virtual void applyConfig(IPropertyTree *options) override
+    {
+        if (!options) return;
+
+        for (int opt=wuOptionFirst; opt<arraysize(wuOptionsDefaults); opt++)
+        {
+             StringBuffer wuOptionName ("@");
+            wuOptionName.append(wuOptionsDefaults[opt].name);
+            __int64 val =  options->getPropInt64(wuOptionName, -1);
+            if (val!=-1)
+                setOptionValue(static_cast<WuOptions>(opt), val);
+        }
+    }
+
+    virtual void applyConfig(IConstWorkUnit * wu) override
+    {
+        for (int opt=0; opt<arraysize(wuOptionsDefaults); opt++)
+        {
+            StringBuffer wuOptionName ("analyzer_");
+            wuOptionName.append(wuOptionsDefaults[opt].name);
+            __int64 val = wu->getDebugValueInt64(wuOptionName, -1);
+            if (val!=-1)
+                setOptionValue(static_cast<WuOptions>(opt), val);
+        }
+    }
+    virtual stat_type queryOption(unsigned opt) const override { return wuOptions[opt]; }
+private:
+    __int64 wuOptions[arraysize(wuOptionsDefaults)];
```

Comment:
this is fine. a more common pattern in our code is to have an enumeration tag on the end for the maximum value.